packages/theme/src/types/Theme.ts

```diff
@@ -133,6 +134,8 @@ export interface Theme extends IScheme {
    */
   components?: ComponentsStyles;
 
+  icons?: IIconSubset;
```

Comment:
I don't think this belongs to the theme. I would prefer to have a separate "provider" which purely serving the purpose of overriding icons - `IconProvider` for providing &` useIcons()` for consuming.

Reply:
A separate IconProvider would also solve the circular dependency issue probably.

Reply:
Another option for addressing the circular dep would be moving IIconSubset out of style-utilities and into some other package (but not sure which one)

Reply:
Actually I take both of those comments back somewhat--going to make another comment on the test file.